src/blenderbim/blenderbim/tool/qto.py

```diff
@@ -35,3 +40,113 @@ def get_radius_of_selected_vertices(cls, obj):
     def set_qto_result(cls, result):
         bpy.context.scene.BIMQtoProperties.qto_result = str(round(result, 3))
+
+    @classmethod
+    def set_active_object(cls, object):
+        bpy.context.view_layer.objects.active = object
+
+    @classmethod
+    def get_pset_qto_object_ifc_info(cls, object):
+        element = tool.Ifc.get_entity(object)
+        pset_qto_ifc_instance = ifcopenshell.util.element.get_psets(element, qtos_only = True)
+        return pset_qto_ifc_instance
+
+    @classmethod
+    def get_pset_qto_properties(cls, object):
+        file = tool.Ifc.get()
+        schema = file.schema
+        pset_qto = util.pset.PsetQto(schema)
+        pset_qto_name = cls.get_pset_qto_name(object)
+        pset_qto_properties = pset_qto.get_by_name(pset_qto_name).get_info()['HasPropertyTemplates']
+        return pset_qto_properties
+
+    @classmethod
+    def get_pset_qto_name(cls, object):
+        applicable_pset_names = cls.get_applicable_pset_names(object)
+        for applicable_pset_name in applicable_pset_names:
+            if 'Qto_' in applicable_pset_name:
+                pset_qto_name = applicable_pset_name
+                return pset_qto_name
+
+    @classmethod
+    def get_applicable_pset_names(cls, object):
+        file = tool.Ifc.get()
+        schema = file.schema
+        pset_qto = util.pset.PsetQto(schema)
+        entity = tool.Ifc.get_entity(object)
+        ifc_object_type = entity.get_info()['type']
+        applicable_pset_names = pset_qto.get_applicable_names(ifc_object_type)
+        return applicable_pset_names
+
+    @classmethod
+    def edit_qto(cls, object, calculated_quantities):
+        file = tool.Ifc.get()
+        pset_qto_id = cls.get_pset_qto_id(object)
+        pset_qto_name = cls.get_pset_qto_name(object)
+
+        ifcopenshell.api.run("pset.edit_qto",
+                file,
+                **{"qto" : pset_qto_id, "name" : pset_qto_name, "properties": calculated_quantities}
+            )
+
+    @classmethod
+    def get_pset_qto_id(cls, object):
+        file = tool.Ifc.get()
+        pset_qto_name = cls.get_pset_qto_name(object)
+        pset_qto_object_ifc_instance = cls.get_pset_qto_object_ifc_info(object)
+        pset_qto_id = file.by_id(pset_qto_object_ifc_instance[pset_qto_name]['id'])
+        return pset_qto_id
+
+    @classmethod
+    def get_pset_qto_name(cls, object):
+        applicable_pset_names = cls.get_applicable_pset_names(object)
+        for applicable_pset_name in applicable_pset_names:
+            if 'Qto_' in applicable_pset_name:
+                pset_qto_name = applicable_pset_name
+                return pset_qto_name
+
+    @classmethod
+    def get_new_quantity(cls, object, quantity_name, alternative_prop_names):
+        calculator = QtoCalculator()
+        new_quantity = calculator.guess_quantity(quantity_name, alternative_prop_names, object)
+        return new_quantity
+
+    @classmethod
+    def get_non_calculated_value(cls):
+        return 0
```

Comment:
Remove this function

Reply:
Ok... i added it just to highlight the possibility that "guess quantity" (or in the future "calculate quantity") doesn't actually calculate the value...